### ISSUE TITLE
 Quick scrolling of the crafting menu

### ISSUE THREAD

**Issue by Hlototo** [anegostudios/VintageStory-Issues#4797]
### Game Version

1.20.0-rc.5

### Platform

Windows

### Modded

Modded

### SP/MP

Singleplayer

### Description

when quickly scrolling through the craft menu, a crash occurs in both single-player and multiplayer

### How to reproduce

при быстрой прокрутке меню крафтов  

### Logs

```
Running on 64 bit Windows 10.0.19045.0 with 16280 MB RAM
Game Version: v1.20.0-rc.5 (Unstable)
25.12.2024 18:05:39: Critical error occurred
Loaded Mods: natshumanskin@1.0.1, primitivesurvival@3.7.4, game@1.20.0-rc.5, creative@1.20.0-rc.5, survival@1.20.0-rc.5
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
   at System.Collections.Generic.List`1.get_Item(Int32 index)
   at Vintagestory.Client.NoObf.TextureAtlasManager.AllocateTextureSpace(Int32 width, Int32 height, Int32& textureSubId, TextureAtlasPosition& texPos) in VintagestoryLib\Client\Render\Atlas\TextureAtlasManager.cs:line 390
   at Vintagestory.Client.NoObf.TextureAtlasManager.InsertTexture(IBitmap bmp, Int32& textureSubId, TextureAtlasPosition& texPos, Single alphaTest) in VintagestoryLib\Client\Render\Atlas\TextureAtlasManager.cs:line 298
   at Vintagestory.Client.NoObf.TextureAtlasManager.GetOrInsertTexture(AssetLocationAndSource loc, Int32& textureSubId, TextureAtlasPosition& texPos, CreateTextureDelegate onCreate, Single alphaTest)
   at Vintagestory.Client.NoObf.TextureAtlasManager.GetOrInsertTexture(CompositeTexture ct, Int32& textureSubId, TextureAtlasPosition& texPos, Single alphaTest) in VintagestoryLib\Client\Render\Atlas\TextureAtlasManager.cs:line 989
   at Vintagestory.GameContent.ShapeTextureSource.get_Item(String textureCode) in VSSurvivalMod\Block\BlockShapeFromAttributes.cs:line 68
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElement(Int32 indexForLogging, MeshData meshdata, ShapeElement element, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 400
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElements(MeshData meshdata, ShapeElement[] elements, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 309
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShapeElements(MeshData meshdata, ShapeElement[] elements, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 309
   at Vintagestory.Client.NoObf.ShapeTesselator.TesselateShape(Shape shapeBase, MeshData& modeldata, Vec3f wholeMeshRotation, Vec3f wholeMeshOffset, Single wholeMeshScale, TesselationMetaData meta) in VintagestoryLib\Client\Render\Tesselation\ShapeTesselator.cs:line 250
   at Vintagestory.GameContent.BlockShapeFromAttributes.GetOrCreateMesh(IShapeTypeProps cprops, ITexPositionSource overrideTexturesource, String overrideTextureCode) in VSSurvivalMod\Block\BlockShapeFromAttributes.cs:line 653
   at Vintagestory.GameContent.BlockShapeFromAttributes.OnBeforeRender(ICoreClientAPI capi, ItemStack itemstack, EnumItemRenderTarget target, ItemRenderInfo& renderinfo) in VSSurvivalMod\Block\BlockShapeFromAttributes.cs:line 496
   at Vintagestory.Client.NoObf.InventoryItemRenderer.GetItemStackRenderInfo(ClientMain game, ItemSlot inSlot, EnumItemRenderTarget target, Single dt) in VintagestoryLib\Client\Render\InventoryItemRenderer.cs:line 315
   at Vintagestory.Client.NoObf.InventoryItemRenderer.RenderItemstackToGui(ItemSlot inSlot, Double posX, Double posY, Double posZ, Single size, Int32 color, Single dt, Boolean shading, Boolean origRotate, Boolean showStackSize) in VintagestoryLib\Client\Render\InventoryItemRenderer.cs:line 194
   at Vintagestory.Client.NoObf.InventoryItemRenderer.RenderItemstackToGui(ItemSlot inSlot, Double posX, Double posY, Double posZ, Single size, Int32 color, Boolean shading, Boolean origRotate, Boolean showStackSize) in VintagestoryLib\Client\Render\InventoryItemRenderer.cs:line 189
   at Vintagestory.Client.NoObf.RenderAPIGame.RenderItemstackToGui(ItemSlot inSlot, Double posX, Double posY, Double posZ, Single size, Int32 color, Boolean shading, Boolean rotate, Boolean showStacksize) in VintagestoryLib\Client\API\RenderAPIGame.cs:line 273
   at Vintagestory.GameContent.GuiHandbookItemStackPage.RenderListEntryTo(ICoreClientAPI capi, Single dt, Double x, Double y, Double cellWidth, Double cellHeight) in VSSurvivalMod\Systems\Handbook\Gui\GuiHandbookItemStackPage.cs:line 91
   at Vintagestory.GameContent.GuiElementFlatList.RenderInteractiveElements(Single deltaTime) in VSSurvivalMod\Systems\Handbook\Gui\GuiElementFlatList.cs:line 194
   at Vintagestory.API.Client.GuiComposer.Render(Single deltaTime) in VintagestoryApi\Client\UI\GuiComposer.cs:line 718
   at Vintagestory.API.Client.GuiDialog.OnRenderGUI(Single deltaTime) in VintagestoryApi\Client\UI\Dialog\GuiDialog.cs:line 389
   at Vintagestory.GameContent.GuiDialogHandbook.OnRenderGUI(Single deltaTime) in VSSurvivalMod\Systems\Handbook\Gui\GuiDialogHandbook.cs:line 546
   at Vintagestory.Client.NoObf.GuiManager.OnRenderFrameGUI(Single deltaTime) in VintagestoryLib\Client\Systems\Gui\GuiManager.cs:line 318
   at Vintagestory.Client.NoObf.ClientEventManager.TriggerRenderStage(EnumRenderStage stage, Single dt) in VintagestoryLib\Client\Util\ClientEventManager.cs:line 188
   at Vintagestory.Client.NoObf.ClientMain.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\ClientMain.cs:line 991
   at Vintagestory.Client.GuiScreenRunningGame.RenderToDefaultFramebuffer(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 242
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 695
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 670
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 88
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 334
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass10_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 130
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93
```

**Comment by Craluminum2413** [comment]
Can't reproduce

**Comment by Hlototo** [comment]
> Can't reproduce

I found a bug, it is caused by one of the mods.

**Comment by radfast** [comment]
It's a duplicate of issue #4693 

This is already fixed in current version 1.20.0-rc.6
